src/netconf_acm.c

```diff
@@ -1622,19 +1666,68 @@ const struct lyd_node *
 ncac_check_diff(const struct lyd_node *diff, const char *user)
 {
     const struct lyd_node *node = NULL;
+    char **groups = NULL;
+    uint32_t group_count;
 
     /* NACM LOCK */
     pthread_mutex_lock(&nacm.lock);
 
+    if (ncac_collect_groups(sr_get_context(np2srv.sr_conn), user, &groups, &group_count)) {
+        goto cleanup;
+    }
+
     /* any node can be used in this case */
     if (!ncac_allowed_tree(diff->schema, user)) {
-        node = ncac_check_diff_r(diff, user, NULL);
+        node = ncac_check_diff_r(diff, user, NULL, groups, group_count);
         if (node) {
             ++nacm.denied_data_writes;
         }
     }
+    ncac_free_groups(groups, group_count);
 
     /* NACM UNLOCK */
+cleanup:
```

Comment:
Similar unlock refactoring. Put freeing the groups into cleanup, too, it is part of it and no reason not to (even though it is not required, for now).